diff --git a/libavcodec/libsvtav1.c b/libavcodec/libsvtav1.c
index 0f9d455660..5b751a215b 100644
--- a/libavcodec/libsvtav1.c
+++ b/libavcodec/libsvtav1.c
@@ -73,7 +73,8 @@ typedef struct SvtContext {
     int tile_rows;
 #endif
     int enc_mode;
-    int crf;
+    float crf;
+    uint8_t extended_crf_qindex_offset;
     int qp;
 } SvtContext;
 
@@ -165,7 +166,7 @@ static int config_enc_params(EbSvtAv1EncConfiguration *param,
         param->look_ahead_distance  = svt_enc->la_depth;
 #endif
 
-    if (svt_enc->enc_mode >= 0)
+    if (svt_enc->enc_mode >= -3)
         param->enc_mode             = svt_enc->enc_mode;
 
     if (avctx->bit_rate) {
@@ -185,7 +186,16 @@ static int config_enc_params(EbSvtAv1EncConfiguration *param,
             FFMAX(avctx->bit_rate, avctx->rc_max_rate);
 
     if (svt_enc->crf > 0) {
-        param->qp                   = svt_enc->crf;
+        float crf_value;
+        uint32_t extended_q_index;
+
+        crf_value = svt_enc->crf;
+        param->qp = FFMIN(MAX_QP_VALUE, (int)crf_value);
+
+        // Calculate the quarter-step offset
+        extended_q_index = (uint32_t)(crf_value * 4);
+        param->extended_crf_qindex_offset = extended_q_index - param->qp * 4;
+
         param->rate_control_mode    = 0;
     } else if (svt_enc->qp > 0) {
         param->qp                   = svt_enc->qp;
@@ -588,7 +598,7 @@ static const AVOption options[] = {
         { "high", NULL, 0, AV_OPT_TYPE_CONST, { .i64 = 1 }, 0, 0, VE, "tier" },
 #endif
     { "preset", "Encoding preset",
-      OFFSET(enc_mode), AV_OPT_TYPE_INT, { .i64 = -1 }, -1, MAX_ENC_PRESET, VE },
+      OFFSET(enc_mode), AV_OPT_TYPE_INT, { .i64 = -4 }, -4, MAX_ENC_PRESET, VE },
 
     FF_AV1_PROFILE_OPTS
 
@@ -620,8 +630,8 @@ static const AVOption options[] = {
         { LEVEL("7.3", 73) },
 #undef LEVEL
 
-    { "crf", "Constant Rate Factor value", OFFSET(crf),
-      AV_OPT_TYPE_INT, { .i64 = 0 }, 0, 63, VE },
+    { "crf", "Constant Rate Factor value (can use quarter-step increments)",
+      OFFSET(crf), AV_OPT_TYPE_FLOAT, { .dbl = 0 }, 0, 70, VE },
     { "qp", "Initial Quantizer level value", OFFSET(qp),
       AV_OPT_TYPE_INT, { .i64 = 0 }, 0, 63, VE },
 #if FF_API_SVTAV1_OPTS
